cmd/charm/charmcmd/plugin.go

```diff
@@ -97,116 +98,235 @@ func pluginHelpTopic() string {
 	} else {
 		longest := 0
 		for _, plugin := range existingPlugins {
-			if len(plugin.name) > longest {
-				longest = len(plugin.name)
+			if len(plugin.Name) > longest {
+				longest = len(plugin.Name)
 			}
 		}
 		for _, plugin := range existingPlugins {
-			fmt.Fprintf(output, "%-*s  %s\n", longest, plugin.name, plugin.description)
+			fmt.Fprintf(output, "%-*s  %s\n", longest, plugin.Name, plugin.Description)
 		}
 	}
 	return output.String()
 }
 
+var PluginDescriptionLastCallReturnedCache bool
+
 // pluginDescriptionsResults holds memoized results for getPluginDescriptions.
 var pluginDescriptionsResults []pluginDescription
 
 // getPluginDescriptions runs each plugin with "--description".  The calls to
 // the plugins are run in parallel, so the function should only take as long
 // as the longest call.
+// But there was a bug reported on this behavior and it is noticably slow
```

Comment:
This sounds like a commit message, not a doc comment.

How about:

    // We cache results in $XDG_CACHE_HOME/charm-command-cache
    // or $HOME/.cache/charm-command-cache if $XDG_CACHE_HOME
    // isn't set, invalidating the cache if executable modification times change.